Add `pl-card` element (#6978)
PrairieLearn/PrairieLearn: docs/elements.md

- [`pl-external-grader-variables`](#pl-external-grader-variables-element): Displays expected and given variables for externally graded questions.
- [`pl-xss-safe`](#pl-xss-safe-element): Removes potentially unsafe code from HTML code.
- [`pl-file-preview`](#pl-file-preview-element): Displays a preview of submitted files.

> - [`pl-card`](#pl-card): Displays content within a card container.

Reviewer: ```suggestion
- [`pl-card`](#pl-card): Displays content within a card styled component.
```

> - [`pl-card`](#pl-card): Displays content within a card-styled component.

Reviewer 2: ```suggestion
- [`pl-card`](#pl-card-element): Displays content within a card-styled component.
```

- [`pl-file-editor` to provide an in-browser code environment](#pl-file-editor-element)

### `pl-card` element

Displays question content inside a card component. Optionally displays content in a header, footer, or image caps via tag attributes.

#### Sample element

```html
<pl-card header="Header" title="Title" width="50%" img-bottom-src="https://picsum.photos/536/354">
  <pl-question-panel> This card is 50% width and has a bottom image cap. </pl-question-panel>
</pl-card>
```

> 

Reviewer: Any chance we could get an image here? I think the one from the PR would be perfect.

Reviewer 2: I think that we're no longer including images in documentation anymore.

Reviewer: Mmm, time for a new issues :)

Reviewer: Added issue #6981

#### Customizations

| Attribute        | Type   | Default | Description                             |
| ---------------- | ------ | ------- | --------------------------------------- |
| `header`         | string | -       | Contents of the card header             |
| `title`          | string | -       | Contents of the card title              |
| `subtitle`       | string | -       | Contents of the card subtitle           |
| `contents`       | string | -       | Raw contents of the card body           |
| `footer`         | string | -       | Contents of the card footer             |
| `img-top-src`    | string | -       | Source URL for the top image cap        |
| `img-bottom-src` | string | -       | Source URL for the bottom image cap     |
| `width`          | string | auto    | Width of the card (25%, 50%, 75%, auto) |

> 

Reviewer: ```suggestion

#### Details

```

- [`pl-file-editor` to provide an in-browser code environment](#pl-file-editor-element)

### `pl-card` element

> Displays question content inside a card component. Optionally displays content in a header, footer, or image caps via tag attributes.

Author: ```suggestion
Displays question content within a card-styled component. Optionally displays content in a header, footer, or image caps via tag attributes.
```

#### Sample element

```html
<pl-card header="Header" title="Title" width="50%" img-bottom-src="https://picsum.photos/536/354">
  <pl-question-panel> This card is 50% width and has a bottom image cap. </pl-question-panel>
</pl-card>
```

#### Customizations

| Attribute        | Type   | Default | Description                             |
| ---------------- | ------ | ------- | --------------------------------------- |
| `header`         | string | -       | Contents of the card header             |
| `title`          | string | -       | Contents of the card title              |
| `subtitle`       | string | -       | Contents of the card subtitle           |
| `contents`       | string | -       | Raw contents of the card body           |
| `footer`         | string | -       | Contents of the card footer             |
| `img-top-src`    | string | -       | Source URL for the top image cap        |
| `img-bottom-src` | string | -       | Source URL for the bottom image cap     |
| `width`          | string | auto    | Width of the card (25%, 50%, 75%, auto) |

#### Details

> These `pl-card` attributes mirror the options of [Bootstrap 4 cards](https://getbootstrap.com/docs/4.6/components/card/).

Author: ```suggestion
The `pl-card` attributes mirror the options of [Bootstrap 4 cards](https://getbootstrap.com/docs/4.6/components/card/).
```

#### Example implementations

> - [element/card]

Reviewer 2: I think you have to add a link to this at the bottom of the page for this to work.

Author: Fixed in 24c4449c802923ab6285eee8c528458967f6c47e

- [`pl-file-editor` to provide an in-browser code environment](#pl-file-editor-element)

### `pl-card` element

> Displays question content within a card-styled component. Optionally displays content in a header, footer, or image caps via tag attributes.

Author: Done 1d0b73c751b66bb6d6f3faefa6e917c46a7244de

#### Sample element

```html
<pl-card
  header="Header"
  title="Title"
  width="50%"
  img-bottom-src="https://via.placeholder.com/720x480"
>
  <pl-question-panel> This card is 50% width and has a bottom image. </pl-question-panel>
</pl-card>
```

#### Customizations

| Attribute        | Type   | Default | Description                             |
| ---------------- | ------ | ------- | --------------------------------------- |
| `header`         | string | -       | Contents of the card header             |
| `title`          | string | -       | Contents of the card title              |
| `subtitle`       | string | -       | Contents of the card subtitle           |
| `contents`       | string | -       | Raw contents of the card body           |
| `footer`         | string | -       | Contents of the card footer             |
| `img-top-src`    | string | -       | Source URL for the top image            |
| `img-top-alt`    | string | -       | Alternative text for the top image      |
| `img-bottom-src` | string | -       | Source URL for the bottom image         |
| `img-bottom-alt` | string | -       | Alternative text for the bottom image   |

> | `width`          | string | auto    | Width of the card (25%, 50%, 75%, auto) |

Reviewer 3: This is how we express enum-ish values in other elements (will require reformatting this table with Prettier):

```suggestion
| `width`          |"25%", "50%", "75%", or "auto" | auto    | Width of the card |
```
